lib/hammer_cli_foreman/compute_profile.rb

```diff
@@ -0,0 +1,423 @@
+      validate_options do
+        any(:option_name,:option_id).required
+      end
+
+      build_options
+    end
+
+    class SetAttributes < HammerCLIForeman::CreateCommand
+      command_name _('set-attributes')
+      resource :compute_attributes
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes.'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('interface parameters, should be comma separated list of values.'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values.'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+
+      extend_help do |h|
+        h.section _('Provider specific options') do |h|
+          ::HammerCLIForeman::ComputeProfile.compute_resources.each do |name, provider|
+            h.section name do |h|
+              if defined?(provider.compute_profile_create_help)
+                provider.compute_profile_create_help(h)
+              else
+                provider.host_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id, :option_compute_profile_name )
+        validator.any(:option_compute_resource_id, :option_compute_resource_name)
+      end
+
+      def request_params
+        params = {}
+        params['compute_attribute'] = {}
+        if option_compute_profile_name
+         params['compute_profile_id'] = HammerCLIForeman.foreman_resource!(:compute_profiles).call(:index)['results'].select { |hash|  hash['name'] == option_compute_profile_name }[0]['id']
+        elsif option_compute_profile_id
+          params['compute_profile_id'] = option_compute_profile_id
+        end
+        if option_compute_resource_name
+          params['compute_resource_id'] = HammerCLIForeman.foreman_resource!(:compute_resources).call(:index)['results'].select{ |hash| hash['name']== option_compute_resource_name }[0]['id']
+        elsif option_compute_resource_id
+          params['compute_resource_id'] = option_compute_resource_id
+        end
+        params['compute_attribute']['vm_attrs'] = option_compute_attributes || {}
+        if option_interface
+          params['compute_attribute']['vm_attrs']['interfaces_attributes'] = {Time.now.to_i.to_s => option_interface}
+        end
+        if option_volume
+            params['compute_attribute']['vm_attrs']['volumes_attributes'] = {Time.now.to_i.to_s => option_volume}
+        end
+        params
+      end
+
+      success_message _('Compute profile attributes are set')
+      failure_message _('Could not set the compute profile attributes')
+      build_options :without => :id
+    end
+
+    class UpdateAttributes < HammerCLIForeman::UpdateCommand
```

Comment:
In the issue was mentioned `erase-attributes` command as well, but it seems like there is no endpoint for that, so I guess there is a possibility to simulate it with modified `UpdateAttributes` command. But I'm not sure for that, maybe @tstrachota has a better opinion.

Reply:
There is also remove-interface command and remove volume command that can help remove it.